### PR TITLE
Add flake.nix

### PR DESCRIPTION
**Pull request by matthewbauer**
This adds a basic flake.nix.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          maintainers = [ maintainers.bennofs ];
+          platforms = platforms.all;
+        };
+      };
+    });
+    checks = forAllSystems (system: {
+      nix-index = self.packages.nix-index.${system};
+    });
+    defaultPackage = forAllSystems (system: self.packages.${system}.nix-index);
+    devShell = forAllSystems (system: with nixpkgsFor.${system}; stdenv.mkDerivation {
```
Comment (teto) [inline flake.nix:43]: wont self.packages.nix-index be enough ?

File: default.nix [hunk 2]
```diff
@@ -1,32 +1,3 @@
...
-
-  meta = with stdenv.lib; {
-    description = "A files database for nixpkgs";
-    homepage = https://github.com/bennofs/nix-index;
-    license = with licenses; [ bsd3 ];
-    maintainers = [ maintainers.bennofs ];
-    platforms = platforms.all;
-  };
-}
+(import (fetchTarball https://github.com/edolstra/flake-compat/archive/master.tar.gz) {
```
Comment (teto) [inline default.nix:1]: you could also add flake-compat as a flake-input so that it is pinned too.

Comment (edude03) [reply default.nix:1]: Yeah definitely use flake-compat as an input

Comment (teto) [reply default.nix:1]: you  can load the flake.lock toml as done in https://github.com/haskell/haskell-language-server/blob/master/shell.nix

**Pull request merged**